File: 조준장/level3/단속카메라.py
def solution(routes):
    answer = 0
    routes.sort(key=lambda x : x[1]) # 나간 지점을 중심으로 정렬
    camera = -30000
    for route in routes:
        
        # 나간 지점을 중심으로 정렬했기 때문에
        # 시작 지점을 비교하여 나간 지점보다 진입 지점이 크다면 새로운 cctv를 설치해야한다. 
        if camera < route[0]: 
            answer += 1
            camera = route[1]
    
    return answer

File: 이훈일/Programmers/Level3/단속카메라.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        Arrays.sort(routes, new Comparator<int[]>(){
            @Override
            public int compare(int[] o1, int[] o2) {
                return o1[0] - o2[0];
            }
        });
        
        int start = routes[0][0];
        int end = routes[0][1];
        int count = 1;
        for(int i = 1; i < routes.length; i++) {
            int carStart = routes[i][0];
            int carEnd = routes[i][1];
            
            if(carStart <= end) {
                start = carStart;
                end = Math.min(carEnd, end);
            } else {
                count++;
                start = carStart;
                end = carEnd;
            }
        }
        
        return count;
    }
}
